qutip/tests/test_random.py

```diff
@@ -31,65 +31,233 @@
 #    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###############################################################################
 
-import scipy.sparse as sp
-import scipy.linalg as la
 import numpy as np
-from numpy.testing import assert_equal, assert_, run_module_suite
-
+import pytest
+from functools import partial
 from qutip import qeye
-from qutip.random_objects import (rand_ket, rand_dm, rand_herm, rand_unitary,
-                                  rand_ket_haar, rand_dm_hs,
-                                  rand_super, rand_unitary_haar, rand_dm_ginibre,
-                                  rand_super_bcsz)
-
-def test_rand_unitary_haar_unitarity():
-    """
-    Random Qobjs: Tests that unitaries are actually unitary.
-    """
-    U = rand_unitary_haar(5)
-    I = qeye(5)
-
-    assert_(U * U.dag() == I)
-
-def test_rand_dm_ginibre_rank():
-    """
-    Random Qobjs: Ginibre-random density ops have correct rank.
-    """
-    rho = rand_dm_ginibre(5, rank=3)
-
-    rank = sum([abs(E) >= 1e-10 for E in rho.eigenenergies()])
-    assert_(rank == 3)
-
-def test_rand_super_bcsz_cptp():
-    """
-    Random Qobjs: Tests that BCSZ-random superoperators are CPTP.
-    """
-    S = rand_super_bcsz(5)
-    assert_(S.iscptp)
-
-def check_func_dims(func, args, kwargs, dims):
+from qutip.random_objects import *
+from qutip.core.data import to
```

Comment:
avoid star-imports, don't import `to` directly

Reply:
Are the changes in this file related to this PR?

Reply:
I only needed 1 new test, but having 2 files for these tests was bugging me...
I will split this in a new PR.

Reply:
Oh, I wouldn't go to the effort of splitting it - could you just tag which tests are really new?

Reply:
`test_random_type` is the only new test.

But I also rewrote them with `pytest.fixture`.
So while it's the same thing that is tested, the mechanic is changed.
`test_random_seeds(rand_func_check)` was 6 tests in the now deleted test_rand.py...

I did do too much when adding 1 test and I thinks is worth to actually push that to master, not dev.major.
(There will be a conflict when merging master->dev.major)
It won't be hard to split the PR, I just have to checkout this one file in a new branch and erase the new test.